### PR TITLE
[Network Drive] Fix connector hanging the framework

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes https://github.com/elastic/connectors-py/issues/2121

This PR aims to wrap synchronous calls inside run_in_executor to prevent connector from hanging the framework.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

File: connectors/sources/network_drive.py [hunk 1]
```diff
@@ -116,9 +116,10 @@ async def _remote_validation(self, advanced_rules):
                 validation_message=e.message,
             )
 
-        self.source.create_connection()
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(executor=None, func=self.source.create_connection)
```
Comment (artem-shelkovnikov) [inline connectors/sources/network_drive.py:120]: You used `asyncio.to_thread` in different PR, is there any difference?

Comment (praveen-elastic) [reply connectors/sources/network_drive.py:120]: nope, they are the same. to_thread uses run_in_executor inside it 

image

Comment (artem-shelkovnikov) [reply connectors/sources/network_drive.py:120]: Can you change this PR to also do `to_thread` for consistency then?

File: connectors/sources/network_drive.py [hunk 2]
```diff
@@ -341,14 +341,18 @@ def create_connection(self):
             port=self.port,
         )
 
-    @cached_property
```
Comment (artem-shelkovnikov) [inline connectors/sources/network_drive.py:344]: Will this have performance impact? Is it used multiple times throughout the sync?

Comment (artem-shelkovnikov) [reply connectors/sources/network_drive.py:344]: The impact is pretty big, seems like, but I feel like it's okay to merge this PR and address it later.

**Pull request merged**